metagraph/src/annotation/binary_matrix/multi_brwt/clustering.cpp

```diff
@@ -225,9 +267,42 @@ Partition greedy_matching(const std::vector<sdsl::bit_vector> &columns,
     return partition;
 }
 
-LinkageMatrix
-agglomerative_greedy_linkage(std::vector<sdsl::bit_vector>&& columns,
-                             size_t num_threads) {
+void merge(const sdsl::bit_vector &first, sdsl::bit_vector *second) {
+    assert(second);
+    assert(first.size() == second->size());
+    *second |= first;
+}
+
+void merge(const SparseColumn &first, SparseColumn *second) {
+    assert(second);
+
+    const auto &[size_first, col_first] = first;
+    const auto &[size_second, col_second] = *second;
+
+    SparseColumn merged;
+    auto &[size_merged, col_merged] = merged;
+    size_merged = std::min(size_first, size_second);
+    col_merged.reserve(col_first.size() + col_second.size());
+
+    auto first_end = std::lower_bound(col_first.begin(), col_first.end(), size_merged);
+    auto second_end = std::lower_bound(col_second.begin(), col_second.end(), size_merged);
```

Comment:
```suggestion
    auto first_end = size_first == size_merged
        ? col_first.end()
        : std::lower_bound(col_first.begin(), col_first.end(), size_merged);
    auto second_end = size_second == size_merged
        ? col_second.end()
        : std::lower_bound(col_second.begin(), col_second.end(), size_merged);
```
can't you avoid this call in some cases?

Reply:
I think it's not worth it. It's the last value, so the bin-search will immediately find it.